iiasa/message_ix
# A tutorial for creating Westeros multi-node and modeling different trade possibilities

This PR adds a new tutorial to demonstrate how a multi-node energy system can be built using the MESSAGEix framework. It shows:
- how new nodes can be added to a model and how data can be populated for those new nodes
- how a trade hub can be represented
- how bilateral trade can be modeled
- role of the `message_ix` index set "mode" in representing trade links

## How to review

- Check out this branch and run the tutorial, and make sure it works without a flaw (error or infeasibility).
- Check the results shown in the tutorial and make sure it makes sense and the text describing the results are correct.

## PR checklist
- [x] Add or expand tests; coverage checks both ✅ (added this tutorial to those being tested in `test_tutorials.py`)
